### PR TITLE
refactoring: base multipart methods

### PR DESCRIPTION
**Pull request by jeetiss**

### CODE REVIEW

File: test/api/multipart/multipartComplete.test.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import multipartComplete from '../../../src/api/multipartComplete'
+import * as factory from '../../_fixtureFactory'
+import { getSettingsForTesting } from '../../_helpers'
+import { UploadClientError } from '../../../src/errors/errors'
+
+
+describe('API - multipartComplete', () => {
```
Comment (bautrukevich) [inline test/api/multipart/multipartComplete.test.ts]: How about tests?

File: src/api/request/request.node.ts [hunk 2]
```diff
@@ -106,7 +126,7 @@ const request = (params: RequestOptions): Promise<RequestResponse> => {
             reject(err)
           })
 
-          if (data && (data instanceof Readable || isFormData)) {
+          if (isReadeble(data, isFormData)) {
```
[the remark this answers is not in the document]
Comment (jeetiss) [reply src/api/request/request.node.ts]: fixed

File: test/api/multipart/multipartComplete.test.ts (hunk 1 above)
Comment (jeetiss) [reply test/api/multipart/multipartComplete.test.ts]: fixed

File: test/api/multipart/multipartComplete.test.ts [hunk 3]
```diff
@@ -1,56 +1,78 @@
...
+  fileSize: number,
+  chunkSize: number
+): Buffer | Blob => {
+  const start = chunkSize * index
+  const end = Math.min(start + chunkSize, fileSize)
+
+  return file.slice(start, end)
+}
+
+const naiveMultipart = (file, parts, options) => Promise.all(
```
Comment (bautrukevich) [inline test/api/multipart/multipartComplete.test.ts]: naiveMultipart → nativeMultipart ?

File: test/api/multipart/multipartComplete.test.ts [hunk 4]
```diff
@@ -1,56 +1,78 @@
...
+  it('should be able to complete upload data', async () => {
+    const file = factory.file(11)
+    const settings = getSettingsForTesting({
+      publicKey: factory.publicKey('multipart'),
+      contentType: 'application/octet-stream'
+    })
+    const { uuid: completedUuid, parts } = await multipartStart(file.size, settings)
 
-  //   await multipartUpload(fileToUpload, parts, settings)
+    await naiveMultipart(file, parts, settings)
```
Comment (bautrukevich) [inline test/api/multipart/multipartComplete.test.ts]: naiveMultipart → nativeMultipart?

Comment (jeetiss) [reply test/api/multipart/multipartComplete.test.ts]: nope
`naive` -> `наивный`

Comment (bautrukevich) [reply test/api/multipart/multipartComplete.test.ts]: ok : )

**Pull request merged**